File: javascript/atbash-cipher/atbash-cipher.js
const a = 'a'.charCodeAt(0), z = 'z'.charCodeAt(0);

export const decode = (s) =>
	s.toLowerCase().replace(/[^a-z0-9]/g, '').replace(/[a-z]/g, (c) =>
		String.fromCharCode(z - c.charCodeAt(0) + a)
	);

export const encode = (s) => decode(s).replace(/.{5}(?=.)/g, '$& ')

File: python/atbash-cipher/atbash_cipher.py
from string import ascii_lowercase

TRANTAB = str.maketrans(ascii_lowercase, ascii_lowercase[::-1])


def encode(s):
    alnum = "".join([c for c in s.lower() if c.isalnum()])
    spaced = ' '.join(alnum[i:i + 5] for i in range(0, len(alnum), 5))
    return spaced.translate(TRANTAB)


def decode(string):
    return string.replace(" ", "").translate(TRANTAB)
